src/concrete_intersection.jl

```diff
@@ -176,6 +174,148 @@ function intersection(x::Interval{N},
+    new_bound = b / a
+    if is_leq
+        Y = Interval(close_bound, new_bound)
+    else
+        Y = Interval(new_bound, close_bound)
+    end
+    return Y
+end
+
+# symmetric method
+function intersection(hs::HalfSpace{N}, X::Interval{N}
+                     )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+    return intersection(X, hs)
+end
+
+"""
+    intersection(X::Interval{N}, hp::Hyperplane{N}
+                )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
+
+Compute the intersection of an interval and a hyperplane.
+
+### Input
+
+- `X`  -- interval
+- `hp` -- hyperplane
+
+### Output
+
+If the sets do not intersect, the result is the empty set.
+Otherwise the result is the singleton that describes the intersection.
+"""
+function intersection(X::Interval{N}, hp::Hyperplane{N}
+                     )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
+    @assert dim(hp) == 1 "cannot take the intersection between an interval " *
+                         "and a $(dim(hp))-dimensional hyperplane"
+
+    # a one-dimensional hyperplane is just a point
+    p = hp.b / hp.a[1]
+    if _leq(min(X), p) && _leq(p, max(X))
+        return Singleton([p])
+    else
+        return EmptySet{N}()
+    end
+end
+
+# symmetric method
+function intersection(hp::Hyperplane{N}, X::Interval{N}
+                     )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
+    return intersection(X, hp)
+end
+
+"""
+    intersection(X::Interval{N}, Y::LazySet{N}
+                )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+
+Compute the intersection of an interval and a convex set.
+
+### Input
+
+- `X` -- interval
+- `Y` -- convex set
+
+### Output
+
+If the sets do not intersect, the result is the empty set.
+Otherwise the result is the interval that describes the intersection.
+"""
+function intersection(X::Interval{N}, Y::LazySet{N}
+                     )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+    lower = max(min(X), -ρ(N[-1], Y))
+    upper = min(max(X), ρ(N[1], Y))
+    if lower > upper
+        return EmptySet{N}()
+    else
+        return Interval(lower, upper)
+    end
+end
+
+# symmetric method
+function intersection(Y::LazySet{N}, X::Interval{N}
+                     )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
+    return intersection(X, Y)
+end
+
+# disambiguation
+function intersection(X::Interval{N}, H::AbstractHyperrectangle{N}
+                     )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+    return invoke(intersection, Tuple{Interval{N}, LazySet{N}}, X, H)
+end
+function intersection(H::AbstractHyperrectangle{N}, X::Interval{N}
+                     )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+    return invoke(intersection, Tuple{Interval{N}, LazySet{N}}, X, H)
+end
+function intersection(X::Interval{N}, S::AbstractSingleton{N}
+                     )::Union{Interval{N}, EmptySet{N}} where {N<:Real}
+    return invoke(intersection, Tuple{typeof(S), LazySet{N}}, S, X)
```

Comment:
```suggestion
    return invoke(intersection, Tuple{Interval{N}, LazySet{N}}, S, X)
```

Reply:
To note that there is mismatch between the return type of the function in lines 310-312 (expects to return Interval or EmptySet) and the return type of the function that is invoked (expects Singleton or EmptySet).

Reply:
I think the `intersection` method for a `Singleton` is more efficient.
But what is wrong is is the output type (should be `Union{Singleton{N}, EmptySet{N}}`).